datafusion/functions-window/src/row_number.rs

```diff
@@ -31,8 +31,8 @@ use datafusion_expr::{Expr, PartitionEvaluator, Signature, Volatility, WindowUDF
 
 /// Create a [`WindowFunction`](Expr::WindowFunction) expression for
 /// `row_number` user-defined window function.
-pub fn row_number(args: Vec<Expr>) -> Expr {
-    Expr::WindowFunction(WindowFunction::new(row_number_udwf(), args))
+pub fn row_number() -> Expr {
```

Comment:
Note row_number() does not take an argument -- it was erroniously changed here https://github.com/apache/datafusion/pull/12030 and it happens we didn't have other tests for it

```sql
> select row_number() OVER ();
+-----------------------------------------------------------------------+
| ROW_NUMBER() ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING |
+-----------------------------------------------------------------------+
| 1                                                                     |
+-----------------------------------------------------------------------+
1 row(s) fetched.
Elapsed 0.042 seconds.
```

Reply:
Since the signature is any(0), I guess it fails even if we pass non-empty args to the function.

Reply:
Given there is signature Any(0) but the test in round_trip still works.

I guess we should add `data_types_with_window_udf` somewhere

Reply:
> I guess we should add data_types_with_window_udf somewhere

I don't understand this comment 🤔